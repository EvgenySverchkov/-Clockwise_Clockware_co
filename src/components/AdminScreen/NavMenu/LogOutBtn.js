import React from "react";
import { useDispatch } from "react-redux";
import { Link } from "react-router-dom";
import { toogleAuth } from "../../../store/adminPanel/services/actions";

const LogOutBtn = () => {
  const dispatch = useDispatch();

  function handler() {
    sessionStorage.removeItem("token");
    dispatch(toogleAuth(false));
  }
  return (
    <div className="mr-1">
      <Link
        to="/admin"
        onClick={handler}
        className="btn btn-secondary"
        aria-haspopup="true"
        aria-expanded="false"
      >
        LogOut
      </Link>
    </div>
  );
};

export default LogOutBtn;
