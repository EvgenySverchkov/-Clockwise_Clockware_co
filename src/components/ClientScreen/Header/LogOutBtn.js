import React from "react";
import { Link } from "react-router-dom";

function LogOutBtn({ handler }) {
  return (
    <li className="nav-item">
      <Link className="nav-link" to="/client" onClick={handler}>
        Logout
      </Link>
    </li>
  );
}

export default LogOutBtn;
