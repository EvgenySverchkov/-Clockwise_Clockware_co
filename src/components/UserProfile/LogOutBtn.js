import React from "react";
import { useDispatch, useSelector } from "react-redux";
import { Link } from "react-router-dom";

import { addCurrentOrderToState } from "../../store/ordersManagement/actions";
import { toogleAuthClient } from "../../store/auth/actions";

function LogOutBtn() {
  const state = useSelector((state) => {
    return { currentOrder: state.ordersReducer.currentOrder };
  });
  const dispatch = useDispatch();

  function logOutHandl() {
    ["user", "token"].forEach((item) => localStorage.removeItem(item));
    dispatch(addCurrentOrderToState({ ...state.currentOrder, email: "" }));
    dispatch(toogleAuthClient(false));
  }
  return (
    <Link className="btn btn-danger" to="/" onClick={logOutHandl}>
      Logout
    </Link>
  );
}

export default LogOutBtn;
