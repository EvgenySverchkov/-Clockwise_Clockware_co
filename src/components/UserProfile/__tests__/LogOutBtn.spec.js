import React from "react";
import {create, act} from "react-test-renderer";
import {MemoryRouter} from "react-router-dom";
import configMockStore from "redux-mock-store";
import {Provider} from "react-redux";

import LogOutBtn from "../LogOutBtn";

const mockStore = configMockStore();

describe("Test of <LogOutBtn/>", ()=>{
    const store = mockStore({
        ordersReducer: {
            currentOrder: {}
        }
    });
    it("Default view", ()=>{
        const component = create(
            <MemoryRouter initialEntries={["/"]}>
                <Provider store={store}>
                    <LogOutBtn/>
                </Provider>
            </MemoryRouter>
        );
        expect(component.toJSON()).toMatchSnapshot();
    });
    it("When click on button", ()=>{
        const mockFn = jest.fn()
        global.localStorage.__proto__.removeItem = mockFn;
        const component = create(
            <MemoryRouter initialEntries={["/"]}>
                <Provider store={store}>
                    <LogOutBtn />
                </Provider>
            </MemoryRouter>
        );
        act(()=>{
            component.root.findByProps({className: "btn btn-danger"}).props.onClick();
        });
        expect(mockFn.mock.calls.length).toBe(2);
    });
});