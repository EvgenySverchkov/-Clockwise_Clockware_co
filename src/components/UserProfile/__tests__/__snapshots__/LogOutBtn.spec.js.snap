// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test of <LogOutBtn/> Default view 1`] = `
<a
  className="btn btn-danger"
  href="/"
  onClick={[Function]}
>
  Logout
</a>
`;
